src/PowerShellEditorServices.Protocol/Server/DebugAdapter.cs

```diff
@@ -435,7 +437,28 @@ protected void Stop()
                 // will block the debug adapter initialization process.  The
                 // InitializedEvent will be sent as soon as the RunspaceChanged
                 // event gets fired with the attached runspace.
-                int runspaceId = attachParams.RunspaceId > 0 ? attachParams.RunspaceId : 1;
+
+                int runspaceId = 1;
+                if (string.IsNullOrEmpty(attachParams.LocalRunspaceId))
+                {
+                    runspaceId = attachParams.RunspaceId > 0 ? attachParams.RunspaceId : 1;
+                }
+                else if (int.TryParse(attachParams.LocalRunspaceId, out int localRunspaceId))
```

Comment:
Seems like we could `int.TryParse()` on a string typed RunspaceId.  In the normal "attach" case we'd just set it to "1".  Your new debug config would set it to whatever the user picks from the runspace id pick list.

Reply:
Sweet. Works for me. I will update these PRs sometime this weekend.